pyre/extra_tests/snippets/stdlib_posix.py

```
@@ -33,3 +38,197 @@ def __int__(self):
+    __fspath__ = property(lambda self: None)
+
+
+assert os.fspath(FsPathOk()) == "fspath-ok"
+assert os.fspath(FsPathProperty()) == "fspath-property"
+
+for cls in (FsPathNone, FsPathNoneOverride, FsPathPropertyNone):
+    try:
+        os.fspath(cls())
+    except TypeError as exc:
+        # The object is named, not the `None` that was found in its place.
+        assert cls.__name__ in str(exc), (cls, exc)
```

Ruff (0.16.1)

[warning] Found assertion on exception `exc` in `except` block, use `pytest.raises()` instead

(PT017)